RIOT-OS/RIOT
# treewide: modify Kconfig symbol prefixes

### Contribution description
It was proposed by @aabadie to modify the current prefix conventions for Kconfig symbols while we are in migration, to reduce possible ambiguities. After some [discussion](https://github.com/RIOT-OS/RIOT/pull/14626#discussion_r475079120), the proposal is the following:

| Current prefix | New prefix | Symbol description |
| ------------------ | -------------- | -------------------------- |
| `MODULE_` | `USEMODULE_` | Symbols generated from `USEMODULE` variable. Needed only during migration. |
| `KCONFIG_MODULE_` | `KCONFIG_USEMODULE_` | Symbols used in `menuconfig`s to allow optional configuration of used modules. Depend on `USEMODULE_` symbols. |
| `PKG_` | `USEPKG_` | Symbols generated from `USEPKG` variable. Needed only during migration. |
| `KCONFIG_PKG_` | `KCONFIG_USEPKG_` | Symbols used in `menuconfig`s to allow optional configuration of used packages. Depend on`USEPKG_` symbols. |
| `MOD_` | `MODULE_` | Symbols that model modules in Kconfig. |
| -- (no packages modelled yet) | `PACKAGE_` | Symbols that model packages in Kconfig. |

### Testing procedure
- Check that the naming convention makes sense and is understandable.
- Configurations should be shown and applied as usual (the `USEMODULE` and `USEPKG` variables should generate the same symbols which activate the visibility of configurations).
- Check that all symbols have been renamed and documentation has been updated, according the table above.
- Green `CI`.

### Issues/PRs references
Suggested in #14626
